include/zephyr/kernel.h

```diff
@@ -4569,6 +4569,29 @@ int k_msgq_cleanup(struct k_msgq *msgq);
  */
 __syscall int k_msgq_put(struct k_msgq *msgq, const void *data, k_timeout_t timeout);
 
+/**
+ * @brief Send a message to front of a message queue.
+ *
+ * This routine sends a message to message queue @a q.
```

Comment:
Docs don't actually say what the function does, should be something like: "This routine sends a message to message queue q, inserting it before any other messages already in the queue."

Reply:
Done.